projects/components/src/table/data/table-cdk-data-source-api.ts

```diff
@@ -5,6 +5,7 @@ import { TableDataSource } from './table-data-source';
 
 export interface TableDataSourceProvider {
   data?: TableDataSource<TableRow>;
+  mode: TableMode;
```

Comment:
The widget already knows the mode and when it changes right ? Can we remove the changes to the actual table component/data source here (except for the styling ones)?